eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -1309,7 +1318,10 @@ registerPool = do
 
 -- | Performs the action without running the fix-up function on any transactions
 withNoFixup :: ImpTestM era a -> ImpTestM era a
-withNoFixup = local $ iteDoTxFixupL .~ False
+withNoFixup = local $ iteFixupL .~ pure
+
+withPostFixup :: (Tx era -> ImpTestM era (Tx era)) -> ImpTestM era a -> ImpTestM era a
+withPostFixup f = local $ iteFixupL %~ (f >=>)
```

Comment:
Your implementation is actually a `withPreFixup` :smile: It should be `(>=> f)`

My suggestion is to add one `withCustomFixup` that can be exported and would be useful for modification of the fixup function in any way desired. 
```suggestion
withPostFixup f = withCustomFixup (>=> f)

withCustomFixup :: 
  ((Tx era -> ImpTestM era (Tx era)) -> Tx era -> ImpTestM era (Tx era))
  -> ImpTestM era a
  -> ImpTestM era a
withCustomFixup f = local $ iteFixupL %~ f
```

Here is an equivalent implementation for `withNoFixup`, but I don't think it is any simpler than the current implementation:
```haskell
withNoFixup = withCustomFixup (const pure)
```

Reply:
Yes! I like the `withCustomFixup`, that solves my concern about this, thank you!